### PR TITLE
add pagination, add tests for pagination, fix other tests

### PR DESCRIPTION
**Pull request by Lex-23**
- write and set up pagination for GET `list salesdashboard` and GET `list offers`
- fix tests after pagination
- add tests for pagination

### CODE REVIEW

File: cryptostock/account/tests/test_get_list_offers.py [hunk 1]
```diff
@@ -183,39 +183,73 @@ def test_get_list_offers_db_calls_from_client(auth_client, client_account):
     OfferFactory.create_batch(100, client=client_account)
 
     with CaptureQueriesContext(connection) as query_context:
-        response = auth_client.get("/api/offer/")
+        response = auth_client.get("/api/offer/?limit=100")
 
     assert response.status_code == 200
-    assert len(query_context) == 5
+    assert len(query_context) == 6
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_offers.py:189]: Why has it increased?

File: cryptostock/account/tests/test_get_list_offers.py [hunk 2]
```diff
@@ -19,12 +20,12 @@ def test_client_get_list_offers(auth_client, client_account):
     offer2 = OfferFactory(deal=sale, count=d.Decimal("30.0456"), client=offer1.client)
     OfferFactory(deal=sale, client=ClientFactory())
 
-    response = auth_client.get("/api/offer/")
+    response = auth_client.get("/api/offer/?limit=10")
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_offers.py:23]: Do we have default limit value?

Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py:23]: Added one request everywhere due to pagination. (Calculate all count of items - count(*))

File: cryptostock/account/tests/test_get_list_offers.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py:189]: Noup. Is it necessary?

Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py:189]: Added one request everywhere due to pagination. (Calculate all count of items - count(*))

File: cryptostock/account/tests/test_get_list_offers.py (hunk 2 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py:23]: Noup. Is it necessary?

Comment (vadim-denisov) [reply cryptostock/account/tests/test_get_list_offers.py:23]: Yes.

File: cryptostock/account/tests/test_get_list_salesdashboard.py [hunk 3]
```diff
@@ -59,15 +60,49 @@ def test_get_list_sales_dashboard_db_calls(auth_user, broker_account):
...
+@pytest.mark.parametrize("limit,offset1,offset2", [(10, 0, 10), (20, 20, 40)])
+def test_get_list_sales_dashboard_pagination_check_offset(
+    auth_user, limit, offset1, offset2
+):
+    SalesDashboardFactory.create_batch(100)
+
+    response = auth_user.get(f"/api/salesdashboard/?limit={limit}&offset={offset1}")
+
+    assert response.status_code == 200
+    sale1_id = response.json()["results"][0]["id"]
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_salesdashboard.py:100]:
```suggestion
    sale_id = response.json()["results"][0]["id"]
```

Comment (vadim-denisov) [reply cryptostock/account/tests/test_get_list_salesdashboard.py:100]: You don't have sale2 here.

File: cryptostock/account/tests/test_get_list_salesdashboard.py [hunk 4]
```diff
@@ -59,15 +60,49 @@ def test_get_list_sales_dashboard_db_calls(auth_user, broker_account):
...
+def test_get_list_sales_dashboard_pagination_check_limit(
+    auth_user, count, limit, offset
+):
+    SalesDashboardFactory.create_batch(count)
+
+    response = auth_user.get(f"/api/salesdashboard/?limit={limit}&offset={offset}")
+
+    assert response.status_code == 200
+    assert len(response.json()) == 4
+    assert tuple(response.json().keys()) == ("count", "next", "previous", "results")
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_salesdashboard.py:86]: Looks like line 85 in enough.
```suggestion
    assert tuple(response.json().keys()) == ("count", "next", "previous", "results")
```

File: cryptostock/account/tests/test_get_list_offers.py [hunk 5]
```diff
@@ -205,15 +205,49 @@ def test_get_list_offers_db_calls_from_broker(auth_broker, broker_account):
...
+@pytest.mark.parametrize("limit,offset1,offset2", [(10, 0, 10), (20, 20, 40)])
+def test_get_offers_pagination_check_offset(
+    auth_client, client_account, limit, offset1, offset2
+):
+    OfferFactory.create_batch(100, client=client_account)
+
+    response = auth_client.get(f"/api/offer/?limit={limit}&offset={offset1}")
+
+    assert response.status_code == 200
+    offer1_id = response.json()["results"][0]["id"]
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_offers.py:245]:
```suggestion
    offer_id = response.json()["results"][0]["id"]
```

File: cryptostock/account/tests/test_get_list_offers.py (hunk 2 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py]: fixed

File: cryptostock/account/tests/test_get_list_salesdashboard.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_salesdashboard.py:100]: got it

File: cryptostock/account/tests/test_get_list_salesdashboard.py (hunk 4 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_salesdashboard.py:86]: ok

File: cryptostock/account/tests/test_get_list_offers.py (hunk 5 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_offers.py:245]: got it

File: cryptostock/account/tests/test_get_list_salesdashboard.py [hunk 6]
```diff
@@ -59,15 +60,48 @@ def test_get_list_sales_dashboard_db_calls(auth_user, broker_account):
...
+)
+def test_get_list_sales_dashboard_pagination_check_limit(
+    auth_user, count, limit, offset
+):
+    SalesDashboardFactory.create_batch(count)
+
+    response = auth_user.get(f"/api/salesdashboard/?limit={limit}&offset={offset}")
+
+    assert response.status_code == 200
+    assert tuple(response.json().keys()) == ("count", "next", "previous", "results")
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_get_list_salesdashboard.py:85]: Keys can have different order between tests. Use set:
```suggestion
    assert response.json().keys() == {"count", "next", "previous", "results"}
```

Comment (Lex-23) [reply cryptostock/account/tests/test_get_list_salesdashboard.py:85]: got it

**Review by vadim-denisov — APPROVED** [review]

**Comment by Lex-23** [comment]
[Add pagination for api](https://trello.com/c/HsZaF6GN/15-add-pagination-for-api)

**Pull request merged**